packages/frontend2/src/app/(legacy)/scaling-next/projects/[slug]/opengraph-image.tsx

```diff
@@ -0,0 +1,134 @@
+import { join } from 'path'
+import { layer2s, layer3s } from '@l2beat/config'
+import { readFile, readdir } from 'fs/promises'
+import { ImageResponse } from 'next/og'
+
+const websiteURL =
+  process.env.NODE_ENV === 'production'
+    ? 'https://www.l2beat.com/'
+    : 'http://localhost:3000/'
+
+const size = {
+  width: 1200,
+  height: 630,
+}
+
+const scalingProjects = [...layer2s, ...layer3s]
+
+export async function generateStaticParams() {
+  return scalingProjects.map((project) => ({
+    slug: project.display.slug,
+  }))
+}
+
+export async function generateImageMetadata({ params }: Props) {
+  return [
+    {
+      id: params.slug,
+      size,
+      alt: `Project page for ${params.slug}`,
+      contentType: 'image/png',
+      runtime: 'nodejs',
+    },
+  ]
+}
+
+interface Props {
+  params: {
+    slug: string
+  }
+}
+
+export default async function Image({ params }: Props) {
+  const project = scalingProjects.find((p) => p.display.slug === params.slug)
+  if (!project) throw new Error('Project not found')
```

Comment:
I guess we shouldn't throw an error here since this will result in error 500. It's better to `return new NextResponse(null, { status: 404 })`.